---
title: "Variant Analysis with maftools"
author: "Eileen Owens"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(dev = "png",
                      dpi = 300,
                      echo = TRUE,
                      cache = TRUE,
                      cache.lazy = FALSE,
                      warning = FALSE,
                      keep_md = TRUE)
```

# Introduction
The purpose of this script is to perform variant analysis on maf files with maftools. The maftools documentation is available here: <https://www.bioconductor.org/packages/release/bioc/vignettes/maftools/inst/doc/maftools.html>

```{r instructions, results='hide', include=FALSE}
# BEORE KNITTING THIS RMARKDOWN, make sure all code chunks have the options you desire in their header:
# eval=FALSE --> this code chunk will not be run
# include=FALSE --> code chunk will be run but not displayed in the final document
# echo=FALSE --> will run the code but will not display it in the code chunk above its results in the final document
# results='hide' --> code will be run and shown in the final document, but its results will not be displayed
# message=FALSE --> code will be run and shown in the final document, but no messages generated by the code will be displayed
```

# Software
```{r install, eval=FALSE}
# First time only
BiocManager::install("maftools")
```

```{r package-load, results='hide', message=FALSE}
# Load every time
library(knitr)
library(maftools)
library(dplyr)
library(VennDiagram)
library(ggplot2)
```

# Data input
Input: One concatenated .maf file of all 96 canine CD4+ PTCL samples and one concatenated .maf file of all control samples (5 samples of sorted nodal CD4+ T cells and 2 samples of sorted CD4+ thymocytes). These input files were generated from RNA-seq data with the GATK pipeline. SNPs and INDELs were filtered based on these recommendations for hard filtering from the GATK documentation: <https://gatk.broadinstitute.org/hc/en-us/articles/360035890471-Hard-filtering-germline-short-variants>. SNPs with QD2 \< 2, SOR \> 3, FS \> 60, MQ \< 40, MQRankSum \< -12.5, and ReadPosRankSum \< -8 and INDELs with QD \< 2, FS \> 200, and ReadPosRankSum \< -20 were excluded. After filtering, vcf files were annotated with the Ensembl Variant Effect Predictor (VEP) and converted to maf files with vcf2maf. Variants annotated as intronic, splice_site, splice_region, or having low or mediator effects were also filtered out and excluded.

```{r data-input, message=FALSE, results='hide'}
# set working directory
setwd("C:/Users/edlarsen/Documents/PTCLRNASeq")

# read PTCL maf file
mymaf <- read.maf(maf = 'Cohort_2/Input/AllPTCLs.CanFam31.QD2.vep.refiltered.maf')

# read CTRL maf file
mymaf_ctrl <- read.maf(maf = 'Cohort_2/Input/AllCTRLs.CanFam31.QD2.vep.refiltered.maf')
```

## Inspect maf objects and export a summary
### PTCLs
```{r sample-summary}
# Print a table of the 20 samples with the highest number of variants
sampleSummary <- getSampleSummary(mymaf)
first20Samples <- head(sampleSummary, 20)
kable(first20Samples, caption = "Top 20 Canine CD4 PTCL Samples with Highest Number of Variants")

# Print a table of the top 20 mutated genes
geneSummary <- getGeneSummary(mymaf)
topVariantGenes <- head(geneSummary, 20)
kable(topVariantGenes, caption = "Top 20 Mutated Genes in Canine CD4 PTCL")

# Write maf summary to an output file
write.mafSummary(maf = mymaf, basename = 'Cohort_2/Output/CD4PTCL_maftools')
```

### Controls
```{r sample-summary-ctrl}
# Print a table of the 20 samples with the highest number of variants
sampleSummary_ctrl <- getSampleSummary(mymaf_ctrl)
first20Samples_ctrl <- head(sampleSummary_ctrl, 20)
kable(first20Samples_ctrl, caption = "Top 20 Canine CD4 Control Samples with Highest Number of Variants")

# Print a table of the top 20 mutated genes
geneSummary_ctrl <- getGeneSummary(mymaf_ctrl)
topVariantGenes_ctrl <- head(geneSummary_ctrl, 20)
kable(topVariantGenes_ctrl, caption = "Top 20 Mutated Genes in Canine CD4 Controls")

# Write maf summary to an output file
write.mafSummary(maf = mymaf_ctrl, basename = 'Cohort_2/Output/CD4CTRL_maftools')
```

# Initial data visualization
```{r}
# set colors to annotate mutation types
var_cols = RColorBrewer::brewer.pal(n = 9, name = 'Paired')
names(var_cols) = c(
  'In_Frame_Ins',
  'Missense_Mutation',
  'In_Frame_Del',
  'Frame_Shift_Ins',
  'Translation_Start_Site',
  'Nonstop_Mutation',
  'Frame_Shift_Del',
  'Multi_Hit',
  'Nonsense_Mutation'
)

titvcols = RColorBrewer::brewer.pal(n = 6, name = 'Set3')
names(titvcols) = c("C>T", "C>G", "C>A", "T>A", "T>C", "T>G")
```


## Plotting MAF summaries
Displays the number of variants in each sample as a stacked barplot and variant types as a boxplot summarized by Variant_Classification. 
### PTCLs
```{r plotmafsummary, fig.width = 10, fig.height = 6}
plotmafSummary(maf = mymaf,
               color = var_cols,
               titvColor = titvcols,
               rmOutlier = TRUE, 
               addStat = 'median', 
               dashboard = TRUE, 
               titvRaw = FALSE)
```

### Controls
```{r plotmafsummary-ctrl, fig.width = 10, fig.height = 6}
plotmafSummary(maf = mymaf_ctrl,
               color = var_cols,
               titvColor = titvcols,
               rmOutlier = TRUE, 
               addStat = 'median', 
               dashboard = TRUE, 
               titvRaw = FALSE)
```


## Barplots of mutated genes
### PTCLs
```{r mafbarplot-ptcl, fig.width=8}
par(mar = c(5, 0.1, 4, 2))

mafbarplot(
  mymaf,
  color = var_cols,
  n = 20,
  genes = NULL,
  fontSize = 0.6,
  includeCN = FALSE,
  legendfontSize = 1,
  borderCol = "#34495e",
  showPct = TRUE
)
```

### Controls
```{r mafbarplot-ctrl, fig.width=8}
par(mar = c(5, 0.1, 4, 2))

mafbarplot(
  mymaf_ctrl,
  color = var_cols,
  n = 20,
  genes = NULL,
  fontSize = 0.6,
  includeCN = FALSE,
  legendfontSize = 1,
  borderCol = "#34495e",
  showPct = TRUE
)
```

## Oncoplot of genes commonly mutated in human PTCL
### Canine CD4+ PTCLs
```{r humanPTCLgenes-PTCL, message=FALSE, fig.height = 8}

hPTCLgenes = c("TET2", "DNMT3A", "PTEN", "TP53", "CDKN2A", "MYC", "STAT3", "BCL11B", "BCL6", "CD244", "CD247", "FASLG", "TP63", "TPRG1", "FYN", "IBTK", "LATS1", "ZC3H12D", "TNFAIP3")

# subset PTCL maf for only fusion gene partners
mymaf_hPTCL <- subsetMaf(mymaf, genes = hPTCLgenes)

# draw oncoplot
oncoplot(
  maf = mymaf_hPTCL,
  genes = hPTCLgenes,
  colors = var_cols,
  titleText = "Canine CD4+ PTCL Variants in Genes Commonly Mutated in Human PTCL"
)
```

### Canine CD4+ control lymphocytes and thymocytes
```{r humanPTCLgenes-CTRL, message=FALSE, fig.height=8}
# subset control maf for only fusion gene partners
mymaf_CTRL_hPTCL <- subsetMaf(mymaf_ctrl, genes = hPTCLgenes)

# draw oncoplot
oncoplot(
  maf = mymaf_CTRL_hPTCL,
  genes = hPTCLgenes,
  colors = var_cols,
  titleText = "Canine CD4+ CTRL Variants in Genes Commonly Mutated in Human PTCL"
)
```

## Oncoplot of genes commonly mutated in canine PTCL
### PTCLs
```{r k9ptclgenes, message=FALSE}
# define list of genes
genes <- c("PTEN", "SATB1", "MAP2K1", "NLRP14", "KCND2", "PSMA1", "TBC1D26")

# subset PTCL maf for only these genes
mymaf_PTCLgenes <- subsetMaf(mymaf, genes=genes)

# draw oncoplot
oncoplot(
  maf = mymaf_PTCLgenes,
  genes = genes,
  titleText = "Canine CD4+ PTCL Variants in Genes Commonly Mutated in Canine TCL"
)
```

### Controls
```{r k9ptclgenes-ctrl, message=FALSE}
# subset control maf for only these genes
mymaf_ctrl_PTCLgenes <- subsetMaf(mymaf_ctrl, genes=genes)

# draw oncoplot
oncoplot(
  maf = mymaf_ctrl_PTCLgenes,
  genes = genes,
  titleText = "Canine CD4+ CTRL Variants in Genes Commonly Mutated in Canine TCL"
)
```

## Oncoplot of canine PTCL fusion gene partners
### PTCLs
```{r fusiongenes-PTCL, message=FALSE}
# define list of fusion gene partners
fusiongenes = c("GATD3A", "LMO4", "PTMA", "NCL", "JPT1", "MROH1", "TPD52L2", "TOX2", "REV3L", "FYN", "HMGB1", "BZW1", "HSPD1", "CHD3", "PER1", "EIF5A", "GRB10", "IKZF1", "MYC", "TRIB1", "YWHAZ", "KLF10", "SRSF5")

# subset PTCL maf for only fusion gene partners
mymaf_PTCLfusion <- subsetMaf(mymaf, genes = fusiongenes)

# draw oncoplot
oncoplot(
  maf = mymaf_PTCLfusion,
  genes = fusiongenes,
  titleText = "Canine CD4+ PTCL Variants Called in Fusion Partner Genes"
)
```

### Controls
```{r fusiongenes-CTRL, message=FALSE}
# subset control maf for only fusion gene partners
mymaf_CTRLfusion <- subsetMaf(mymaf_ctrl, genes = fusiongenes)

# draw oncoplot
oncoplot(
  maf = mymaf_CTRLfusion,
  genes = fusiongenes,
  titleText = "Canine CD4+ CTRL Variants Called in Fusion Partner Genes"
)
```

## Transition and Transversions
Boxplot summarizes the overall distribution of different conversions, and stacked barplot shows fraction of conversions in each sample.
### PTCLs
```{r titv}
mymaf.titv = titv(maf = mymaf,
                  plot = FALSE,
                  useSyn = TRUE)
# plot titv summary
plotTiTv(res = mymaf.titv)
```

### Controls
```{r titv-ctrl}
mymaf_ctrl.titv = titv(maf = mymaf_ctrl,
                  plot = FALSE,
                  useSyn = TRUE)
# plot titv summary
plotTiTv(res = mymaf_ctrl.titv)
```

# Compare variants called in both tumor and control samples
## Venn Diagram
```{r vennDiagram, message=FALSE, warning=FALSE, error=FALSE, results='hide'}
tumor_vars <- mymaf@data
ctrl_vars <- mymaf_ctrl@data

tumor_relevant_cols <- tumor_vars[, c("Hugo_Symbol", "Variant_Classification", "Tumor_Sample_Barcode")]
#tumor_relevant_cols <- unique(tumor_relevant_cols)
control_relevant_cols <- ctrl_vars[, c("Hugo_Symbol", "Variant_Classification", "Tumor_Sample_Barcode")]
#control_relevant_cols <- unique(control_relevant_cols)

# identify variants that exist in both tumor and control
matching_vars <- merge(tumor_relevant_cols, control_relevant_cols, 
                           by = c("Hugo_Symbol", "Variant_Classification"),
                       allow.cartesian = TRUE)


##### Venn Diagram #####

# Read in list of genes and variants in both groups
varPTCL <- unique(paste(tumor_relevant_cols$Hugo_Symbol, tumor_relevant_cols$Variant_Classification))
varCTRL <- unique(paste(control_relevant_cols$Hugo_Symbol, control_relevant_cols$Variant_Classification))

# Prepare color palette
library(RColorBrewer)
myCol <- brewer.pal(8, "Pastel2")[c(1, 2)]

# Create venn diagram of overlapping variants between the groups
set.seed(1)
venn1 <- venn.diagram(
  x = list(varPTCL, varCTRL),
  category.names = c("CD4+ PTCL", "CD4+ CTRL"),
  
  # Output features
  filename = NULL,
  disable.logging = TRUE,
  
  # Title
  main = "Variants Shared Between CD4+ PTCL and \nControl CD4+ Lymphocytes and Thymocytes",
  main.cex = 1.5,
  main.fontfamily = "sans",
  main.fontface = "bold",

  # Circles
  fill = c(alpha("#440154ff", 0.3), alpha('#21908dff', 0.3)),
  lwd = 1,
  col = c("#440154ff", '#21908dff'),
  
  # Numbers
  cex=1.5,
  fontfamly = "sans",
  
  # Categories
  cat.cex = 1.5,
  cat.fontfamily = "sans",
  cat.fontface = "bold",
  cat.dist = c(0.05, 0.05),
  cat.pos = c(-27, 27),
  cat.default.pos = "outer",
  cat.col = c("#440154ff", '#21908dff'),
  scaled = FALSE,
)
grid.newpage()
grid.draw(venn1)
```

## Filter genes from tumor samples that were called in control samples
```{r fitlervenn, message=FALSE, results='hide'}
#mymaf_filtered <- filterMaf(mymaf, genes=varCTRL)

# Filter tumor maf based on both gene and variant type
mymaf_filtered <- mymaf@data[!(paste(Hugo_Symbol, Variant_Classification) %in%
                                 paste(matching_vars$Hugo_Symbol, matching_vars$Variant_Classification))]

write.table(mymaf_filtered, file = "ptcl_unique_vars_only.maf", sep = "\t", quote = FALSE, row.names = FALSE)
mymaf_filtered <- read.maf("ptcl_unique_vars_only.maf")
```

# Data visualization after filtering
Data visualization of only those genes that were mutated in tumor samples, and not in control samples.

## Plotting MAF summary in PTCLs
Displays the number of variants in each PTCL sample as a stacked barplot and variant types as a boxplot summarized by Variant_Classification.
```{r plotmafsummary-filtered, fig.width = 10, fig.height = 6}
plotmafSummary(maf = mymaf_filtered,
               color = var_cols,
               titvColor = titvcols,
               rmOutlier = TRUE, 
               addStat = 'median', 
               dashboard = TRUE, 
               titvRaw = FALSE)
```

## Barplot of mutated genes in PTCLs
```{r mafbarplot-filtered, fig.width = 8}
par(mar = c(5, 0.1, 4, 2))
mafbarplot(
  mymaf_filtered,
  n = 20,
  genes = NULL,
  color = var_cols,
  fontSize = 0.7,
  includeCN = FALSE,
  legendfontSize = 1,
  borderCol = "#34495e",
  showPct = TRUE
)
```

## Oncoplots
### Top tumor-specific mutated genes
Oncoplot for the top 20 mutated genes after filtering out genes also called in control samples. Note: Variants annotated as Multi_Hit are those genes which are mutated more than once in the same sample.
```{r oncoplot-filtered, fig.height=8, fig.width=10}
#par(mar = c(5, 2, 4, 2))
oncoplot(maf = mymaf_filtered,
         fontSize = 0.5,
         top = 20,
         colors = var_cols,
         titleText = "Top Canine CD4+ PTCL-Specific Variants")
```

### Oncoplot of oncogenic signaling pathway genes in PTCLs
```{r oncoplotsignaling-filtered, results='hide', message=FALSE, fig.height=8, fig.width=10}
oncoplot(maf = mymaf_filtered,
         colors = var_cols,
         pathways = "sigpw",
         titleText = "Top 10 Mutated Oncogenic Signaling Pathways in Canine CD4+ PTCL",
         gene_mar = 8, 
         fontSize = 0.8, 
         topPathways = 10,
         collapsePathway = TRUE)
```

```{r oncoplot-topsignaling-filtered, results='hide', message=FALSE, fig.height=16, fig.width=16}
oncoplot(maf = mymaf_filtered,
         colors = var_cols,
         pathways = "sigpw",
         titleText = "Details of Top Mutated Oncogenic Signaling Pathway in Canine CD4+ PTCL",
         gene_mar = 8, 
         fontSize = 0.8, 
         topPathways = 1)
```


## Oncoplot of genes commonly mutated in human PTCL
```{r humanPTCLgenes-filtered, results='hide', message=FALSE, fig.height = 8}
# draw oncoplot
oncoplot(
  maf = mymaf_filtered,
  genes = hPTCLgenes,
  colors = var_cols,
  titleText = "Canine CD4+ PTCL-Specific Variants in Genes Commonly Mutated in Human PTCL"
)
```

### Oncoplot of commonly mutated genes in canine PTCL
```{r k9ptclgenes-filtered, fig.width=8}
oncoplot(
  maf = mymaf_filtered,
  genes = genes,
  titleText = "Canine CD4+ PTCL-Specific Variants in Genes Commonly Mutated in Canine TCL"
)
```

### Oncoplot of fusion gene partners in PTCLs
```{r fusiongenes-filtered, fig.height=8}
oncoplot(
  maf = mymaf_filtered,
  genes = fusiongenes,
  titleText = "Canine CD4+ PTCL-Specific Variants Called in Fusion Partner Genes"
)
```

## Transition and Transversions
Boxplot summarizes the overall distribution of different conversions, and stacked barplot shows fraction of conversions in each sample.

```{r titv-filtered}
mymafFiltered.titv = titv(maf = mymaf_filtered,
                  plot = FALSE,
                  useSyn = TRUE)
# plot titv summary
plotTiTv(res = mymafFiltered.titv)
```


## Rainfall plots of the 5 tumors with the most mutations
Visualizes hypermutated genomic regions in cancer genomes by plotting inter variant distance on a linear genomic scale. "Kataegis" are defined as those genomic segments containing 6 or more consecutive mutations with an average inter-mutation distance of less than or equal to 1,00 bp 5. If tsb = NULL, the most mutated sample is plotted.
```{r rainfall-filtered, message=FALSE, fig.width=12}
top5 <- head(sampleSummary, 5)
top5 <- as.character(top5$Tumor_Sample_Barcode[1:5])
top5

for (barcode in top5){
  rainfallPlot(maf = mymaf_filtered,
             detectChangePoints = TRUE,
             tsb = barcode,
             width = 10,
             height = 5,
             pointSize = 0.8)
}
```


# Analysis
## Somatic interactions
The somaticInteractions function performs pair-wise Fisher’s Exact test to detect signfiicant pairs of mutually exclusive or co-occurring sets of genes.
```{r fig.width=12, message=FALSE, results='hide'}
#exclusive/co-occurance event analysis on top 25 tumor-specific mutated genes. 
somaticInteractions(maf = mymaf_filtered, 
                    top = 25, 
                    pvalue = c(0.05, 0.1),
                    fontSize = 0.6)
```


# Citations
```{r}
sessionInfo()
citation()
```
