---
title: "Variant-Analysis-with-maftools"
author: "Eileen Owens"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(dev = "png",
                      dpi = 300,
                      echo = TRUE,
                      cache = TRUE,
                      cache.lazy = FALSE,
                      warning = FALSE,
                      keep_md = TRUE)
```

# Introduction
The purpose of this script is to perform variant analysis on maf files with maftools. The maftools documentation is available here: https://www.bioconductor.org/packages/release/bioc/vignettes/maftools/inst/doc/maftools.html

```{r instructions, results='hide', include=FALSE}
# BEORE KNITTING THIS RMARKDOWN, make sure all code chunks have the options you desire in their header:
# eval=FALSE --> this code chunk will not be run
# include=FALSE --> code chunk will be run but not displayed in the final document
# echo=FALSE --> will run the code but will not display it in the code chunk above its results in the final document
# results='hide' --> code will be run and shown in the final document, but its results will not be displayed
# message=FALSE --> code will be run and shown in the final document, but no messages generated by the code will be displayed
```

# Load software package
```{r install, eval=FALSE}
# First time only
BiocManager::install("maftools")
```

```{r package-load, results='hide', message=FALSE}
# Load every time
library(knitr)
library(maftools)
library(dplyr)
```

# Data input
Input: One concatenated .maf file of all samples.
```{r data-input}
# set working directory
setwd("/path/to/working/directory") # If you are a Windows user, you may have to swap the direction of your slashes from "\" to "/"

# read maf file
mymaf <- read.maf(maf = '/path/to/concatenated/maf/file')
```

Inspect maf object and export a summary:
```{r sample-summary}
# Print a table of the 20 samples with the highest number of variants
sampleSummary <- getSampleSummary(mymaf)
first20Samples <- head(sampleSummary, 20)
kable(first20Samples)

# Print a table of the top 20 mutated genes
geneSummary <- getGeneSummary(mymaf)
topVariantGenes <- head(geneSummary, 20)
kable(topVariantGenes)

# Write maf summary to an output file
write.mafSummary(maf = mymaf, basename = '/path/to/output/directory/Date_mafSummary')
```

# Visualization
## Plotting MAF summary
Displays the number of variants in each sample as a stacked barplot and variant types as a boxplot summarized by Variant_Classification.
```{r plotmafsummary}
plotmafSummary(maf = mymaf, 
               rmOutlier = TRUE, 
               addStat = 'median', 
               dashboard = TRUE, 
               titvRaw = FALSE)
```
## Barplot of mutated genes
```{r mafbarplot}
mafbarplot(
  mymaf,
  n = 50,
  genes = NULL,
  color = NULL,
  fontSize = 0.7,
  includeCN = FALSE,
  legendfontSize = 0.7,
  borderCol = "#34495e",
  showPct = TRUE
)
```

## Filter frequently called variants
Remove all genes from the maf object that were called in 100% of the samples.
```{r removeGenes, results='hide'}
# filter the geneSummary data frame to include only those called in all samples
subset_geneSummary <- geneSummary[geneSummary$MutatedSamples==96, ] # Replace '96' with the total number of samples in your maf file
# create a vector of names of the genes called in all samples
geneList <- subset_geneSummary$Hugo_Symbol 
# use filterMaf tool to exclude these genes from the maf object
mymaf_filtered <- filterMaf(mymaf, genes = geneList, )
```

## Oncoplot
Oncoplot for the top ten mutated genes. Note: Variants annotated as Multi_Hit are those genes which are mutated more than once in the same sample.
```{r oncoplot}
oncoplot(maf = mymaf_filtered, top = 20)
```

## Oncoplot of oncogenic signaling pathway genes
```{r}
oncoplot(maf = mymaf_filtered, pathways = "sigpw", gene_mar = 8, fontSize = 0.6, topPathways = 2)
```

## Oncoplot of genes of interest
```{r fusiongenes}
geneList = c("Gene1", "Gene2", "Gene3", "Gene4", "Gene5")
oncoplot(
  maf = mymaf_filtered,
  genes = geneList,
)
```
## Transition and Transversions
Boxplot summarizes the overall distribution of different conversions, and stacked barplot shows fraction of conversions in each sample.
```{r titv}
mymaf.titv = titv(maf = mymaf,
                  plot = FALSE,
                  useSyn = TRUE)
# plot titv summary
plotTiTv(res = mymaf.titv)
```

## Rainfall plot
Visualizes hypermutated genomic regions in cancer genomes by plotting inter variant distance on a linear genomic scale. "Kataegis" are defined as those genomic segments containing 6 or more consecutive mutations with an average inter-mutation distance of less than or equal to 1,00 bp 5.
```{r rainfall}
rainfallPlot(maf = mymaf,
             detectChangePoints = TRUE,
             pointSize = 0.4)
```

## Plotting VAF
```{r vafplot, eval=FALSE}
plotVaf(maf = mymaf)
```

# Analysis
## Somatic interactions
The somaticInteractions function performs pair-wise Fisherâ€™s Exact test to detect signfiicant pairs of mutually exclusive or co-occurring sets of genes.
```{r}
#exclusive/co-occurance event analysis on top 10 mutated genes. 
somaticInteractions(maf = mymaf, top = 25, pvalue = c(0.05, 0.1))
```

## Detecting cancer driver genes based on positional clustering
The concept behind the oncodrive algorithm is that most of the variants in cancer causing genes are enriched at a few specific loci ("hot spots"). This method takes advantage of such positions to identify cancer genes. Citation: Lawrence MS, Stojanov P, Mermel CH, Robinson JT, Garraway LA, Golub TR, Meyerson M, Gabriel SB, Lander ES, Getz G: Discovery and saturation analysis of cancer genes across 21 tumour types. Nature 2014, 505:495-501.
```{r eval=FALSE}
mymaf.sig = oncodrive(maf = mymaf, AACol = 'Protein_Change', minMut = 5, pvalMethod = 'zscore')

# plot results
plotOncodrive(res = mymaf.sig, fdrCutOff = 0.1, useFraction = TRUE, labelSize = 0.5)
```

# Citations
```{r}
sessionInfo()
citation()
```

