---
title: "Variant-Analysis-with-maftools"
author: "Eileen Owens"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(dev = "png",
                      dpi = 300,
                      echo = TRUE,
                      cache = TRUE,
                      cache.lazy = FALSE,
                      warning = FALSE,
                      keep_md = TRUE)
```

# Introduction
The purpose of this script is to perform variant analysis on maf files with maftools. The maftools documentation is available here: https://www.bioconductor.org/packages/release/bioc/vignettes/maftools/inst/doc/maftools.html

```{r instructions, results='hide', include=FALSE}
# BEORE KNITTING THIS RMARKDOWN, make sure all code chunks have the options you desire in their header:
# eval=FALSE --> this code chunk will not be run
# include=FALSE --> code chunk will be run but not displayed in the final document
# echo=FALSE --> will run the code but will not display it in the code chunk above its results in the final document
# results='hide' --> code will be run and shown in the final document, but its results will not be displayed
# message=FALSE --> code will be run and shown in the final document, but no messages generated by the code will be displayed
```

# Load software package
```{r install, eval=FALSE}
# First time only
BiocManager::install("maftools")
install.packages("knitr")
install.packages("dplyr")
install.packages("VennDiagram")
install.packages("ggplot2")
```

```{r package-load, results='hide', message=FALSE}
# Load every time
library(knitr)
library(maftools)
library(dplyr)
library(VennDiagram)
library(ggplot2)
```

# Data input

Input: One concatenated .maf file of all tumor samples and one concatenated .maf file of all control samples.

```{r data-input}
# set working directory
setwd("/path/to/working/directory") # If you are a Windows user, you may have to swap the direction of your slashes from "\" to "/"

# read tumor maf file
mymaf <- read.maf(maf = '/path/to/concatenated/tumor/maf/file')

# read control maf file
mymaf_ctrl <- read.maf(maf = '/path/to/concatenated/control/maf/file')
```


## Inspect maf object and export a summary:

### Tumors

```{r sample-summary}
# Print a table of the 20 samples with the highest number of variants
sampleSummary <- getSampleSummary(mymaf)
first20Samples <- head(sampleSummary, 20)
kable(first20Samples)

# Print a table of the top 20 mutated genes
geneSummary <- getGeneSummary(mymaf)
topVariantGenes <- head(geneSummary, 20)
kable(topVariantGenes)

# Write maf summary to an output file
write.mafSummary(maf = mymaf, basename = '/path/to/output/directory/Date_mafSummary_tumors')
```

### Controls

```{r sample-summary-ctrl}
# Print a table of the 20 samples with the highest number of variants
sampleSummary_ctrl <- getSampleSummary(mymaf_ctrl)
first20Samples_ctrl <- head(sampleSummary_ctrl, 20)
kable(first20Samples_ctrl)

# Print a table of the top 20 mutated genes
geneSummary_ctrl <- getGeneSummary(mymaf_ctrl)
topVariantGenes_ctrl <- head(geneSummary_ctrl, 20)
kable(topVariantGenes_ctrl)

# Write maf summary to an output file
write.mafSummary(maf = mymaf_ctrl, basename = 'path/to/output/directory/Date_mafsummary_controls')
```


# Initial data visualization

## Plotting MAF summaries

Displays the number of variants in each sample as a stacked barplot and variant types as a boxplot summarized by Variant_Classification.

### Tumors

```{r plotmafsummary}
plotmafSummary(maf = mymaf, 
               rmOutlier = TRUE, 
               addStat = 'median', 
               dashboard = TRUE, 
               titvRaw = FALSE)
```


### Controls

```{r plotmafsummary-ctrl}
plotmafSummary(maf = mymaf_ctrl, 
               rmOutlier = TRUE, 
               addStat = 'median', 
               dashboard = TRUE, 
               titvRaw = FALSE)
```


## Barplots of mutated genes

### Tumors

```{r mafbarplot}
mafbarplot(
  mymaf,
  n = 50,
  genes = NULL,
  color = NULL,
  fontSize = 0.7,
  includeCN = FALSE,
  legendfontSize = 0.7,
  borderCol = "#34495e",
  showPct = TRUE
)
```

### Controls

```{r mafbarplot-ctrl}
mafbarplot(
  mymaf_ctrl,
  n = 50,
  genes = NULL,
  color = NULL,
  fontSize = 0.7,
  includeCN = FALSE,
  legendfontSize = 0.7,
  borderCol = "#34495e",
  showPct = TRUE
)
```


# Oncoplot of genes of interest

### Tumors

```{r genesofinterest, message=FALSE}
# define list of genes
genes <- c("GENE1", "GENE2", "GENE3")

# subset tumor maf for only these genes
mymaf_geneList <- subsetMaf(mymaf, genes=genes)

# draw oncoplot
oncoplot(
  maf = mymaf_geneList,
  genes = genes,
)
```

### Controls

```{r genesofinterest-ctrl, message=FALSE}
# subset control maf for only these genes
mymaf_ctrl_geneList <- subsetMaf(mymaf_ctrl, genes=genes)

# draw oncoplot
oncoplot(
  maf = mymaf_ctrl_geneList,
  genes = genes,
)
```


## Transition and Transversions

Boxplot summarizes the overall distribution of different conversions, and stacked barplot shows fraction of conversions in each sample.

### Tumors

```{r titv}
mymaf.titv = titv(maf = mymaf,
                  plot = FALSE,
                  useSyn = TRUE)
# plot titv summary
plotTiTv(res = mymaf.titv)
```

### Controls

```{r titv-ctrl}
mymaf_ctrl.titv = titv(maf = mymaf_ctrl,
                  plot = FALSE,
                  useSyn = TRUE)
# plot titv summary
plotTiTv(res = mymaf_ctrl.titv)
```


# Compare variants called in both tumor and control samples

## Venn Diagram

```{r vennDiagram}
# Read in the list of genes with variants in both groups
varTUMOR <- geneSummary$Hugo_Symbol
varCTRL <- geneSummary_ctrl$Hugo_Symbol

# Prepare color palette
library(RColorBrewer)
myCol <- brewer.pal(8, "Pastel2")[c(1, 2)]

# Create venn diagram of overlapping genes between the groups
set.seed(1)
venn1 <- venn.diagram(
  x = list(varTUMOR, varCTRL),
  category.names = c("TUMOR", "CONTROL"),
  
  # Output features
  filename = NULL,
  disable.logging = TRUE,
  
  # Title
  main = "Genes with Variants Called",
 main.cex = 1.5,
 main.fontfamily = "sans",
 main.fontface = "bold",
  
  # Circles
  fill = c(alpha("#440154ff", 0.3), alpha('#21908dff', 0.3)),
  lwd = 1,
  col = c("#440154ff", '#21908dff'),
  
  # Numbers
  cex=1.5,
  fontfamly = "sans",
  
  # Categories
  cat.cex = 1.5,
  cat.fontfamily = "sans",
 cat.fontface = "bold",
  cat.dist = c(0.05, 0.05),
  cat.pos = c(-27, 27),
  cat.default.pos = "outer",
  cat.col = c("#440154ff", '#21908dff'),
  scaled = FALSE,
)
grid.newpage()
grid.draw(venn1)
```

## Filter genes from tumor samples that were called in control samples

```{r fitlervenn, message=FALSE}
mymaf_filtered <- filterMaf(mymaf, genes=varCTRL)
```


# Data visualization after filtering

Data visualization of only those genes that were mutated in tumor samples, and not in control samples.

## Plotting MAF summary in Tumors

Displays the number of variants in each tumor sample as a stacked barplot and variant types as a boxplot summarized by Variant_Classification.

```{r plotmafsummary-filtered}
plotmafSummary(maf = mymaf_filtered, 
               rmOutlier = TRUE, 
               addStat = 'median', 
               dashboard = TRUE, 
               titvRaw = FALSE)
```

## Barplot of mutated genes in tumors

```{r mafbarplot-filtered}
mafbarplot(
  mymaf_filtered,
  n = 50,
  genes = NULL,
  color = NULL,
  fontSize = 0.7,
  includeCN = FALSE,
  legendfontSize = 0.7,
  borderCol = "#34495e",
  showPct = TRUE
)
```


## Oncoplots

### Top tumor-specific mutated genes

Oncoplot for the top 20 mutated genes after filtering out genes also called in control samples. Note: Variants annotated as Multi_Hit are those genes which are mutated more than once in the same sample.

```{r oncoplot-filtered}
oncoplot(maf = mymaf_filtered, top = 20)
```

### Oncoplot of genes of interest

```{r genesofinterest-filtered, message=FALSE}
# define list of genes
genes <- c("GENE1", "GENE2", "GENE3")

# draw oncoplot
oncoplot(
  maf = mymaf_filtered,
  genes = genes,
)
```

## Oncoplot of oncogenic signaling pathway genes

```{r oncoplotsignaling-filtered, message=FALSE, fig.height=12, fig.width=10}
oncoplot(maf = mymaf_filtered, pathways = "sigpw", gene_mar = 8, fontSize = 0.6, topPathways = 2)
```


## Transition and Transversions

Boxplot summarizes the overall distribution of different conversions, and stacked barplot shows fraction of conversions in each sample.

```{r titv-filtered}
mymafFiltered.titv = titv(maf = mymaf_filtered,
                  plot = FALSE,
                  useSyn = TRUE)
# plot titv summary
plotTiTv(res = mymafFiltered.titv)
```


## Rainfall plot

Visualizes hypermutated genomic regions in cancer genomes by plotting inter variant distance on a linear genomic scale. "Kataegis" are defined as those genomic segments containing 6 or more consecutive mutations with an average inter-mutation distance of less than or equal to 1,00 bp 5.

```{r rainfall-filtered, message=FALSE}
rainfallPlot(maf = mymaf_filtered,
             detectChangePoints = TRUE,
             tsb = NULL,
             width = 10,
             height = 5,
             pointSize = 0.4)
```


# Analysis

## Somatic interactions

The somaticInteractions function performs pair-wise Fisherâ€™s Exact test to detect signfiicant pairs of mutually exclusive or co-occurring sets of genes.

```{r}
#exclusive/co-occurance event analysis on top 10 mutated genes. 
somaticInteractions(maf = mymaf_filtered, top = 25, pvalue = c(0.05, 0.1))
```


# Citations

```{r}
sessionInfo()
citation()
```
